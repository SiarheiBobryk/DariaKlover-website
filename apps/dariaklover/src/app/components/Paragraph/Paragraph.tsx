import * as React from 'react';

import Typography from '@mui/material/Typography';

export interface ParagraphProps {
  children?: React.ReactNode;
  component?: React.ElementType;
  variant?:
    | 'h1'
    | 'h2'
    | 'h3'
    | 'h4'
    | 'h5'
    | 'h6'
    | 'subtitle1'
    | 'subtitle2'
    | 'body1'
    | 'body2'
    | 'caption'
    | 'button'
    | 'overline'
    | 'inherit'
    | undefined;
}

const Paragraph = React.forwardRef(function Paragraph(props: ParagraphProps, ref: React.Ref<HTMLParagraphElement>) {
  const { children, component = 'p', variant = 'body1', ...other } = props;

  return (
    <Typography ref={ref} component={component} variant={variant} gutterBottom {...other}>
      {children}
    </Typography>
  );
});

export default Paragraph;
