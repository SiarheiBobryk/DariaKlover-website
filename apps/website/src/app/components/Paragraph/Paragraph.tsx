import { ElementType, ReactNode, Ref, forwardRef } from 'react';

import Typography from '@mui/material/Typography';

export interface ParagraphProps {
  children?: ReactNode;
  component?: ElementType;
  variant?:
    | 'h1'
    | 'h2'
    | 'h3'
    | 'h4'
    | 'h5'
    | 'h6'
    | 'subtitle1'
    | 'subtitle2'
    | 'body1'
    | 'body2'
    | 'caption'
    | 'button'
    | 'overline'
    | 'inherit'
    | undefined;
}

const Paragraph = forwardRef(function Paragraph(props: ParagraphProps, ref: Ref<HTMLParagraphElement>) {
  const { children, component = 'p', variant = 'body1', ...other } = props;

  return (
    <Typography ref={ref} component={component} variant={variant} gutterBottom {...other}>
      {children}
    </Typography>
  );
});

export default Paragraph;
