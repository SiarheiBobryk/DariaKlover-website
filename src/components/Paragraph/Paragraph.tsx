import * as React from 'react';

import Typography from '@mui/material/Typography';

export interface ParagraphProps {
  children?: React.ReactNode;
  component?: React.ElementType;
  variant?:
    | 'h1'
    | 'h2'
    | 'h3'
    | 'h4'
    | 'h5'
    | 'h6'
    | 'subtitle1'
    | 'subtitle2'
    | 'body1'
    | 'body2'
    | 'caption'
    | 'button'
    | 'overline'
    | 'inherit'
    | undefined;
}

function Paragraph(props: ParagraphProps) {
  const { children, component = 'p', variant = 'body1', ...other } = props;

  return (
    <Typography component={component} variant={variant} {...other}>
      {children}
    </Typography>
  );
}

export default Paragraph;
