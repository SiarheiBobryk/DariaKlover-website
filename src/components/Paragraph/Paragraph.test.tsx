import * as React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import { typographyClasses } from '@mui/material/Typography';

import Paragraph from './Paragraph';

describe('<Paragraph />', () => {
  describe('snapshotting', () => {
    it('should render correctly without a content', () => {
      const paragraph = renderer.create(<Paragraph />).toJSON();
      expect(paragraph).toMatchSnapshot();
    });
    it('should render correctly with a content', () => {
      const paragraph = renderer.create(<Paragraph>Lorem Ipsulum</Paragraph>).toJSON();
      expect(paragraph).toMatchSnapshot();
    });
  });

  describe('Elements inspection', () => {
    it('should be the `p` HTML element by default', () => {
      render(<Paragraph>Дарья Кловер</Paragraph>);
      const paragraph = document.querySelector('p');
      expect(paragraph).toBeInTheDocument();
      expect(paragraph).toHaveTextContent('Дарья Кловер');
    });

    it('should be the `i` HTML element', () => {
      render(<Paragraph component="i">Дарья Кловер</Paragraph>);
      const paragraph = document.querySelector('i');
      expect(paragraph).toBeInTheDocument();
      expect(paragraph).toHaveTextContent('Дарья Кловер');
    });

    it('should have the CSS classes from `p` Typography variant by default', () => {
      render(<Paragraph>Дарья Кловер</Paragraph>);
      expect(screen.getByText('Дарья Кловер')).toHaveClass(typographyClasses.body1);
    });

    it('should have the CSS classes from `strong` Typography variant', () => {
      render(<Paragraph variant="caption">Дарья Кловер</Paragraph>);
      expect(screen.getByText('Дарья Кловер')).toHaveClass(typographyClasses.caption);
    });
  });
});
