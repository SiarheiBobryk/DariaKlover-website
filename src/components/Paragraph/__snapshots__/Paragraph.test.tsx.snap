// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paragraph /> snapshotting should render correctly with a content 1`] = `
<p
  className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
>
  Lorem Ipsulum
</p>
`;

exports[`<Paragraph /> snapshotting should render correctly without a content 1`] = `
<p
  className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
/>
`;
